### DISCUSSION TITLE
Allow implicit in with lambda syntax

### DISCUSSION THREAD

**Discussion by ThadHouse** [dotnet/csharplang#1591] [Language Ideas] [answered]
Currently, if you have a delegate with an `in`, `out`, or `ref` parameter, you have to explicitly list all the parameter types in a lambda  for that delegate. With `out` and `ref`, this makes sense and should stay that way. However, with the addition of `in`, I could see its use in lambdas becoming more popular (I'm using it in a new library I am creating). Since in is implicit on the calling side, if a lambda is created for a delegate with an `in` parameter, the `in` type should be allowed to be implicit. If the delegate contains both in and `ref or out`, then the old rules should still apply.

If a function has multiple lambda taking overloads, with one taking an in lambda, and one taking a non in lambda, the same overloads as currently with in value parameters should apply.

**Comment by alrz** [comment]
I don't like that being implicit, but rather, we could just permit the type to be omitted.
```cs
delegate void D(in int i, ref int j);

D d = (in x, ref y) => {};
```

**Comment by ThadHouse** [comment]
That'd be fine too. Just having to fully type out everything is a pain.

**Comment by alrz** [comment, marked as answer]
proposal here: https://github.com/dotnet/csharplang/issues/338
